object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -464,21 +465,31 @@ private void handleAssignFromRHS(
       if (isVarInDefs(newDefs, (LocalVariableNode) lhs)) {
         ImmutableSet<LocalVarWithTree> setContainingLatestAssignmentPair =
             getSetContainingAssignmentTreeOfVar(newDefs, (LocalVariableNode) lhs);
+        LocalVariableNode localPassedAsMCCParam = getLocalPassedAsMustCallChoiceParam(rhs);
         LocalVarWithTree latestAssignmentPair =
             getAssignmentTreeOfVar(newDefs, (LocalVariableNode) lhs);
-        if (setContainingLatestAssignmentPair.size() == 1) {
-          checkMustCall(
-              setContainingLatestAssignmentPair,
-              typeFactory.getStoreBefore(node),
-              "variable overwritten by assignment " + node.getTree());
-          newDefs.remove(setContainingLatestAssignmentPair);
-        } else {
-          ImmutableSet<LocalVarWithTree> newSetContainingLatestAssignmentPair =
-              FluentIterable.from(setContainingLatestAssignmentPair)
-                  .filter(assignment -> assignment.equals(latestAssignmentPair))
-                  .toSet();
-          newDefs.remove(setContainingLatestAssignmentPair);
-          newDefs.add(newSetContainingLatestAssignmentPair);
+        if (localPassedAsMCCParam == null
```

Comment:
add a comment here saying the MCC case is handled below?

Reply:
https://github.com/kelloggm/object-construction-checker/pull/282/commits/2dc62e3a4d37190f4881de233fdaa7fe334c8e53